[AMDGPU] Port AMDGPURewriteUndefForPHI to new pass manager

jwanggit86 opened the pull request:
This patch ports the AMDGPURewriteUndefForPHI pass to the new pass manager. With this, the pass is supported under both the legacy and the new pass managers.

jwanggit86 committed:
[AMDGPU] Port AMDGPURewriteUndefForPHI to new pass manager

This patch ports the AMDGPURewriteUndefForPHI pass to the new pass
manager. With this, the pass is supported under both the legacy and
the new pass managers.

arsenm commented on llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:
```diff
@@ -177,6 +177,20 @@ bool AMDGPURewriteUndefForPHI::runOnFunction(Function &F) {
+  return PreservedAnalyses::all();
+}
+
 FunctionPass *llvm::createAMDGPURewriteUndefForPHIPass() {
```
Should rename all the legacy pass bits to end with "LegacyPass"

jwanggit86 replied:
Should the class "AMDGPURewriteUndefForPHI" be renamed "AMDGPURewriteUndefForPHILegacy" or "AMDGPURewriteUndefForPHILegacyPass"? If the latter, we will end up with a function named "initializeAMDGPURewriteUndefForPHILegacyPassPass".

arsenm replied:
AMDGPURewriteUndefForPHILegacy, some places have always been stuck with "PassPass"

jwanggit86 committed:
Merge branch 'main' of https://github.com/llvm/llvm-project into port-rewrite-undef-to-new-pass-mgr

jwanggit86 committed:
[AMDGPU] Port AMDGPURewriteUndefForPHI to new pass manager

This patch ports the AMDGPURewriteUndefForPHI pass to the new pass
manager. With this, the pass is supported under both the legacy and
the new pass managers.

arsenm commented on llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:
```diff
@@ -170,13 +170,27 @@ bool rewritePHIs(Function &F, UniformityInfo &UA, DominatorTree *DT) {
+PreservedAnalyses
+AMDGPURewriteUndefForPHIPass::run(Function &F, FunctionAnalysisManager &AM) {
+  UniformityInfo &UA = AM.getResult<UniformityInfoAnalysis>(F);
+  DominatorTree *DT = &AM.getResult<DominatorTreeAnalysis>(F);
```
Why address of here?
